File: src/php/thrift/CeCd/Sdk/Core/Services/RpcModuleIf.php
<?php
/**
 * Created by PhpStorm.
 * User: zhangyubo
 * Date: 2019/4/30
 * Time: 15:21
 */

namespace Thrift\CeCd\Sdk\Core\Services;


interface RpcModuleIf
{
    public function getHost() : string;

    public function getPort() : int;

    public function setHost(string $host) : RpcModuleIf;

    public function setPort(int $port) : RpcModuleIf;

    public function getServiceId(): int;

    public function getConfigKey() : string;

    public function getLang() : string;

    public function getClientInterceptor() : string;

    public function setMode(int $actionType) : RpcModuleIf;

    public function getActionMode() : int;

    public function setDebug(bool $isDebug = false) : RpcModuleIf;

    public function getDebug() : bool;

}

File: src/java/src/main/java/com/cecd/sdk/rpc/RpcModuleIf.java
package com.cecd.sdk.rpc;

import com.cecd.sdk.rpc.interceptor.ClientInterceptor;

public interface RpcModuleIf {

    public int getServiceId();

    public String getServiceName();

    public String getHost();

    public RpcModuleIf setHost(String host);

    public int getPort();

    public RpcModuleIf setPort(int port);

    public String getLang();

    public int getTimeout();

    public RpcModuleIf setTimeout(int timeout);

    public void setInterceptor(ClientInterceptor interceptor);

    public ClientInterceptor getInterceptor();
}
